packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -31,163 +30,85 @@ export const svgLightDark: GeneratePlugin = {
 
-      const { lightToken, darkToken } = params;
+      const svgDark = (await flavor.getContents()) as string; // .svg asset's getContents() returns the svg as a string
+      const svgLight  = (await lightFlavor.getContents()) as string;
 
-      const lightSvgo = new Svgo({
-        plugins: [
-          {
-            addAttributesToSVGElement: {
-              attributes: [{display: lightToken ? `var(${lightToken})` : "var(--hue-web-svg-display-light)"}]
-            }
-          },
-          {
-            removeViewBox: false
-          }
-        ],
-        js2svg: { pretty: true, indent: 2 }
-      });
+      if (svgLight) {
+        // create optimized light/dark svg assets
 
-      const darkSvgo = new Svgo({
-        plugins: [
-          {
-            addAttributesToSVGElement: {
-              attributes: [{display: darkToken ? `var(${darkToken})` : "var(--hue-web-svg-display-dark)"}]
-            }
-          },
-          {
-            removeViewBox: false
-          }
-        ],
-        js2svg: { pretty: true, indent: 2 }
-      });
+        // Parse XML from a string into a DOM Document.
+        const doc = new DOMParser();
+        const xmlLight = doc.parseFromString(svgLight, 'image/svg+xml');
+        const xmlDark = doc.parseFromString(svgDark, 'image/svg+xml');
 
-      if (lightFlavorContents) {
-        // create optimized light/dark svg assets
-        const lightAsset = await lightSvgo.optimize(lightFlavorContents); // .svg asset's getContents() returns a string
-        const darkAsset = await darkSvgo.optimize(darkFlavorContents);
-
-        const mixedParentSvg = await generateMixedSvgParent(lightAsset, darkAsset);
-
-        const mixedSvgo = new Svgo({
-          plugins: [
-            {
-              removeViewBox: false
-            }
-          ],
-          js2svg: { pretty: true, indent: 2 }
-        });
+        const darkEl = xmlDark.documentElement;
+        const lightEl = xmlLight.documentElement;
+
+        // clone the outer element of the svg
+        const wrapperEl = lightEl.cloneNode(false);
+
+        // Remove all attributes from the light and dark elements
+        while (darkEl.attributes.length) {
+          darkEl.removeAttribute(darkEl.attributes[0].name);
+        }
+        while (lightEl.attributes.length) {
+          lightEl.removeAttribute(lightEl.attributes[0].name);
+        }
+
+        // set the appropriate display tokens for light and dark displays
+        const { lightToken, darkToken } = params;
+        darkEl.setAttribute('display', (darkToken? `var(${darkToken})` : `var(--svg-display-dark)`));
+        lightEl.setAttribute('display', (lightToken? `var(${lightToken})` : `var(--svg-display-light)`));
 
-        const mixedString = serializeToString(mixedParentSvg);
-        const mixedAsset = await mixedSvgo.optimize(mixedString);
+        // append to the wrapper SVG
+        wrapperEl.appendChild(darkEl);
+        wrapperEl.appendChild(lightEl);
 
```

Comment:
Is the wrapper SVG still getting the `height`, `width`, and `viewbox` properties added?  see my note below were it was added before. Ty